### PR TITLE
perlPackages.WWWCurl: fix build

### PR DESCRIPTION
**Pull request by stigtsp**
Include patch from AUR to fix compatibility with newer libcurl.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by veprbl** [comment]
@GrahamcOfBorg build perl530Packages.WWWCurl perl528Packages.WWWCurl

**Comment by stigtsp** [comment]
Fixed darwin build issue.

File: pkgs/top-level/perl-packages.nix [hunk 1]
```diff
@@ -21324,16 +21324,16 @@ let
...
     };
-    patches = [ ../development/perl-modules/WWW-Curl-4.17-Skip-preprocessor-symbol-only-CURL_STRICTER.patch ];
+    patches = [
+      (fetchpatch {
+        url = "https://aur.archlinux.org/cgit/aur.git/plain/curl-7.71.0.patch?h=perl-www-curl&id=261d84887d736cc097abef61164339216fb79180";
+        sha256 = "1hiw5lkflfa93z5d6k8fnnml0r08c653bbvvb8zx6gcrlbrdalfs";
+        name = "WWWCurl-curl-7.71.0.patch";
+      })
+    ];
+    NIX_CFLAGS_COMPILE = stdenv.lib.optionalString stdenv.isDarwin "-Wno-return-type";
```
Comment (veprbl) [inline pkgs/top-level/perl-packages.nix:21334]:
```suggestion
    CFLAGS = stdenv.lib.optionalString stdenv.cc.isClang "-Wno-return-type";
```
Regarding CFLAGS, see #79303
Regarding the warning, those are driven by compiler frontend, not the platform.

Comment (stigtsp) [reply pkgs/top-level/perl-packages.nix]: Thx :)

Comment (stigtsp) [reply pkgs/top-level/perl-packages.nix]: Hm, that didn't quite work.

Comment (veprbl) [reply pkgs/top-level/perl-packages.nix]: I see rest of pkgs/top-level/perl-packages.nix uses NIX_CFLAGS_*. I thought perl respected CFLAGS, but I guess I was wrong.

**Comment by veprbl** [comment]
@GrahamcOfBorg build perl530Packages.WWWCurl perl528Packages.WWWCurl

File: pkgs/top-level/perl-packages.nix [hunk 2]
```diff
@@ -21324,16 +21324,16 @@ let
       url = "mirror://cpan/authors/id/S/SZ/SZBALINT/WWW-Curl-4.17.tar.gz";
       sha256 = "1fmp9aib1kaps9vhs4dwxn7b15kgnlz9f714bxvqsd1j1q8spzsj";
     };
-    patches = [ ../development/perl-modules/WWW-Curl-4.17-Skip-preprocessor-symbol-only-CURL_STRICTER.patch ];
```
Comment (veprbl) [inline pkgs/top-level/perl-packages.nix:21327]: Hold on, this patch needs to be removed from the nixpkgs tree.

Comment (stigtsp) [reply pkgs/top-level/perl-packages.nix:21327]: Thx again :+1:

**Comment by veprbl** [comment]
@GrahamcOfBorg build perl530Packages.WWWCurl perl528Packages.WWWCurl

**Pull request merged**